soh/soh/Enhancements/randomizer/context.cpp

```diff
@@ -140,7 +140,7 @@ void Context::GenerateLocationPool() {
     if (mSettings->GetOption(RSK_TRIFORCE_HUNT)) {
         AddLocation(RC_TRIFORCE_COMPLETED);
     }
-    AddLocations(StaticData::overworldLocations);
+    AddLocations(StaticData::GetOverworldLocations());
 
     if (mSettings->GetOption(RSK_FISHSANITY).IsNot(RO_FISHSANITY_OFF)) {
```

Comment:
Fishsanity checks are added when it is off, likely because of GetOverworldLocations catching those too. Trying to play with fishanity on both is also crashing but I don't know if that's the same issue or not

Reply:
In that case we might want to change it so that we don't add all overworld locations and instead we add a filter to only add some overworld checks.

Reply:
One of my "Sould probably get done eventually" things was changing it so only enabled checks are in the location pool at all, instead of adding everything and then defaulting the unused things. Originally it was just keeping the spoiler log clean but it would be better than this wierd peicemeal approach. This could also simplify building the item pool if we simply add the vanilla checks of those locations as we go. Perhaps a good starting point would be adding a condition to add the check to the pool to each location.

Reply:
Has this been resolved, or is it going to be a fix later thing?

Reply:
Probably a fix later thing as this PR is already too big.